pkg/imageregistry/imageregistry_test.go

```diff
@@ -0,0 +1,328 @@
+		addToRuntimeObjects bool
+		expectedError       error
+		client              bool
+	}{
+		{
+			name:                "test",
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageRegistry 'imageRegistryObjName' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "argocdtest",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("imageRegistry object argocdtest doesn't exist"),
+			client:              true,
+		},
+		{
+			name:                "argocdtest",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageRegistry Config 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testImageRegistry := generateImageRegistry(testCase.name)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testImageRegistry)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := Pull(testSettings, testCase.name)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError != nil {
+			assert.NotNil(t, err)
+			assert.Equal(t, testCase.expectedError, err)
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testImageRegistry.Name, builderResult.Object.Name)
+		}
+	}
+}
+
+func TestImageRegistryExist(t *testing.T) {
+	testCases := []struct {
+		testImageRegistryConfig *Builder
+		expectedStatus          bool
+	}{
+		{
+			testImageRegistryConfig: buildValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedStatus:          true,
+		},
+		{
+			testImageRegistryConfig: buildInValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedStatus:          false,
+		},
+		{
+			testImageRegistryConfig: buildInValidImageRegistryBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus:          false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testImageRegistryConfig.Exists()
+		assert.Equal(t, exist, testCase.expectedStatus)
+	}
+}
+
+func TestImageRegistryGet(t *testing.T) {
+	testCases := []struct {
+		testImageRegistry *Builder
+		expectedError     error
+	}{
+		{
+			testImageRegistry: buildValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedError:     nil,
+		},
+		{
+			testImageRegistry: buildInValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedError:     fmt.Errorf("the imageRegistry 'name' cannot be empty"),
+		},
+		{
+			testImageRegistry: buildInValidImageRegistryBuilder(clients.GetTestClients(clients.TestClientParams{})),
```

Comment:
Please update it to:
buildValidImageRegistryBuilder(clients.GetTestClients(clients.TestClientParams{})),